browser/components/CodeEditor.js

```diff
@@ -644,6 +618,57 @@ export default class CodeEditor extends React.Component {
     }
   }
 
+  setCodeEditorLintConfig () {
+    const { mode } = this.props
+    const checkMarkdownNoteIsOpening = mode === 'Boost Flavored Markdown'
+
+    return checkMarkdownNoteIsOpening ? {
+      'getAnnotations': this.validatorOfMarkdown,
+      'async': true
+    } : false
+  }
+
+  validatorOfMarkdown (text, updateLinting) {
+    const { customMarkdownLintConfig } = this.props
+    let lintConfigJson
+    try {
+      Jsonlint.parse(customMarkdownLintConfig)
+      lintConfigJson = JSON.parse(customMarkdownLintConfig)
+    } catch (err) {
+      throw err
+    }
+    const lintOptions = {
+      'strings': {
+        'content': text
+      },
+      'config': lintConfigJson
+    }
+
+    return markdownlint(lintOptions, (err, result) => {
+      if (!err) {
+        const foundIssues = []
+        result.content.map(item => {
+          let ruleNames = ''
+          item.ruleNames.map((ruleName, index) => {
+            ruleNames += ruleName
+            if (index === item.ruleNames.length - 1) {
```

Comment:
Nit: IMO it would be more readable if you'd do an inline if like:

```suggestion
            ruleNames += (index === item.ruleNames.length - 1) ? ': ' : '/'
```

Reply:
I didn't notice it. Thank you for an advice!